### PR TITLE
fix: node.js -> Node.js

### PR DESCRIPTION
**Pull request by nschonni**
Manually reviewed since casing is intended for emails and some urls.

### CODE REVIEW

**Comment by Trott** [comment]
@nodejs/website

**Review by nickserv — APPROVED** [review]

**Review by tniessen — APPROVED** [review]

**Review by 3imed-jaberi — APPROVED** [review]

**Review by ryzokuken — APPROVED** [review]
Thanks!

File: locale/en/blog/weekly-updates/weekly-update.2015-05-15.md [hunk 1]
```diff
@@ -19,11 +19,11 @@ If you would like to see the convergence plan in detail, please check it at [jas
 ## Community Updates
 
 * [Performance Showdown](https://raygun.io/blog/2015/05/performance-showdown-node-js-vs-io-js-v2-0-0/): Node.js vs. io.js v2.0.0.
-* Microsoft has [node.js](https://github.com/Microsoft/node) running on their JavaScript VM `Chakra`.
+* Microsoft has [Node.js](https://github.com/Microsoft/node) running on their JavaScript VM `Chakra`.
 * {node,io}.js [benchmark graphs](http://thlorenz.com/benchgraph/).
```
Comment (Aissaoui-Ahmed) [inline locale/en/blog/weekly-updates/weekly-update.2015-05-15.md:23]:
```suggestion
* {Node,io}.js [benchmark graphs](http://thlorenz.com/benchgraph/).
```

File: locale/en/knowledge/advanced/buffers/how-to-use-buffers.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ layout: knowledge-post.hbs
 
 ## Why Buffers?
 
-Pure javascript, while great with unicode-encoded strings, does not handle straight binary data very well. This is fine on the browser, where most data is in the form of strings. However, node.js servers have to also deal with TCP streams and reading and writing to the filesystem, both which make it necessary to deal with purely binary streams of data.
+Pure javascript, while great with unicode-encoded strings, does not handle straight binary data very well. This is fine on the browser, where most data is in the form of strings. However, Node.js servers have to also deal with TCP streams and reading and writing to the filesystem, both which make it necessary to deal with purely binary streams of data.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/buffers/how-to-use-buffers.md]:
```suggestion
Pure JavaScript, while great with unicode-encoded strings, does not handle straight binary data very well. This is fine on the browser, where most data is in the form of strings. However, Node.js servers have to also deal with TCP streams and reading and writing to the filesystem, both which make it necessary to deal with purely binary streams of data.
```

File: locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ myREPL.on('exit', function (code) {
 
 There you have it - spawn the node REPL as a child process, and pipe your stdin and stdout to its stdin and stdout.  Make sure to listen for the child's 'exit' event, too, or else your program will just hang there when the REPL exits.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md]:
```suggestion
There you have it - spawn the Node.js REPL as a child process, and pipe your stdin and stdout to its stdin and stdout.  Make sure to listen for the child's 'exit' event, too, or else your program will just hang there when the REPL exits.
```

File: locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md [hunk 4]
```diff
@@ -40,10 +40,10 @@ The asynchronous version, on the other hand, does not stop time - instead, the c
...
 
 ### Event Emitters
-Event Emitters are another basic idiom in node.js. A constructor is provided in Node.js core: `require('events').EventEmitter`. An Event Emitter is typically used when there will be multiple parts to the response (since usually you only want to call a callback once). For more details, see the article on [EventEmitters](/en/knowledge/getting-started/control-flow/what-are-event-emitters/)
+Event Emitters are another basic idiom in Node.js. A constructor is provided in Node.js core: `require('events').EventEmitter`. An Event Emitter is typically used when there will be multiple parts to the response (since usually you only want to call a callback once). For more details, see the article on [EventEmitters](/en/knowledge/getting-started/control-flow/what-are-event-emitters/)
 
 ### A gotcha with asynchronous code
 A common mistake in asynchronous code with javascript is to write code that does something like this:
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md]:
```suggestion
A common mistake in asynchronous code with JavaScript is to write code that does something like this:
```

File: locale/en/knowledge/getting-started/how-to-debug-nodejs-applications.md [hunk 5]
```diff
@@ -53,7 +53,7 @@ At this point, you will be met with a fairly empty screen with the `scripts`, `p
...
+This is just like most webkit/firebug debuggers. It has a list of all the javascript files (including Node.js core and third party libraries) which you can select and dive into. To stop the interpreter on a specific line, you set a breakpoint by clicking on the number of the desired line. When the execution is frozen, by a breakpoint or by manually pausing interpretation by pressing the pause button, you can check the callstack and examine all the local, closure, and global variables. You can also modify the code to try and fix behavior. Note that when you modify the code through the script tab, it does not get saved to the file, so you will need to transfer the modifications back by hand.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/getting-started/how-to-debug-nodejs-applications.md]:
```suggestion
This is just like most webkit/firebug debuggers. It has a list of all the JavaScript files (including Node.js core and third party libraries) which you can select and dive into. To stop the interpreter on a specific line, you set a breakpoint by clicking on the number of the desired line. When the execution is frozen, by a breakpoint or by manually pausing interpretation by pressing the pause button, you can check the callstack and examine all the local, closure, and global variables. You can also modify the code to try and fix behavior. Note that when you modify the code through the script tab, it does not get saved to the file, so you will need to transfer the modifications back by hand.
```

File: locale/en/blog/weekly-updates/weekly-update.2015-05-29.md [hunk 6]
```diff
@@ -38,9 +38,9 @@ See https://github.com/nodejs/node/labels/confirmed-bug for complete and current
 * [Schism and Reconciliation](https://nodesource.com/blog/was-this-trip-really-necessary) in the Node Community by Rod Vagg.
 * First Node TSC Meeting available on [SoundCloud](https://soundcloud.com/node-foundation/tsc-meeting-2015-05-27).
 * io.js have got a new Benchmarking Working Group [nodejs/benchmarking#1](https://github.com/nodejs/benchmarking/issues/1).
```
Comment (Aissaoui-Ahmed) [inline locale/en/blog/weekly-updates/weekly-update.2015-05-29.md:40]:
```suggestion
* io.js have got a new Benchmarking Working Group [Nodejs/benchmarking#1](https://github.com/nodejs/benchmarking/issues/1).
```

File: locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md [hunk 7]
```diff
@@ -8,7 +8,7 @@ difficulty: 2
 layout: knowledge-post.hbs
 ---
 
-If you've been using node.js for a while, you've definitely run into streams.  HTTP connections are streams,  open files are streams;  stdin, stdout, and stderr are all streams as well.  A 'stream' is node's I/O abstraction - if you feel like you still need to understand them better, you can read more about them [here](https://nodejs.org/api/stream.html#stream_stream).
+If you've been using Node.js for a while, you've definitely run into streams.  HTTP connections are streams,  open files are streams;  stdin, stdout, and stderr are all streams as well.  A 'stream' is node's I/O abstraction - if you feel like you still need to understand them better, you can read more about them [here](https://nodejs.org/api/stream.html#stream_stream).
 
 Streams make for quite a handy abstraction, and there's a lot you can do with them - as an example, let's take a look at stream.pipe, the method used to take a readable stream and connect it to a writeable steam.  Suppose we wanted to spawn a `node` child process and pipe our stdout and stdin to its corresponding stdout and stdin.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md]:
```suggestion
Streams make for quite a handy abstraction, and there's a lot you can do with them - as an example, let's take a look at stream.pipe, the method used to take a readable stream and connect it to a writeable steam. Suppose we wanted to spawn a `node` child process and pipe our stdout and stdin to its corresponding stdout and stdin.
```

File: locale/en/blog/weekly-updates/weekly-update.2015-05-15.md (hunk 1 above)
Comment (richardlau) [reply locale/en/blog/weekly-updates/weekly-update.2015-05-15.md:23]: It's the name of the linked website so should probably stay as-is.

File: locale/en/blog/weekly-updates/weekly-update.2015-05-29.md (hunk 6 above)
Comment (richardlau) [reply locale/en/blog/weekly-updates/weekly-update.2015-05-29.md:40]: The GitHub org is all lowercase.

File: locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md [hunk 8]
```diff
@@ -8,7 +8,7 @@ difficulty: 2
 layout: knowledge-post.hbs
 ---
 
-If you've been using node.js for a while, you've definitely run into streams.  HTTP connections are streams,  open files are streams;  stdin, stdout, and stderr are all streams as well.  A 'stream' is node's I/O abstraction - if you feel like you still need to understand them better, you can read more about them [here](https://nodejs.org/api/stream.html#stream_stream).
+If you've been using Node.js for a while, you've definitely run into streams.  HTTP connections are streams,  open files are streams;  stdin, stdout, and stderr are all streams as well.  A 'stream' is node's I/O abstraction - if you feel like you still need to understand them better, you can read more about them [here](https://nodejs.org/api/stream.html#stream_stream).
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md]:
```suggestion
If you've been using Node.js for a while, you've definitely run into streams. HTTP connections are streams, open files are streams; stdin, stdout, and stderr are all streams as well. A 'stream' is node's I/O abstraction - if you feel like you still need to understand them better, you can read more about them [here](https://nodejs.org/api/stream.html#stream_stream).
```

File: locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md [hunk 9]
```diff
@@ -36,7 +36,7 @@ myREPL.on('exit', function (code) {
 
 There you have it - spawn the node REPL as a child process, and pipe your stdin and stdout to its stdin and stdout.  Make sure to listen for the child's 'exit' event, too, or else your program will just hang there when the REPL exits.
 
-Another use for stream.pipe is file streams.  In node.js, fs.createReadStream and fs.createWriteStream are used to create a stream to an open file descriptor.  Now let's look at how one might use stream.pipe to write to a file.  You'll probably recognize most of the code:
+Another use for stream.pipe is file streams.  In Node.js, fs.createReadStream and fs.createWriteStream are used to create a stream to an open file descriptor.  Now let's look at how one might use stream.pipe to write to a file.  You'll probably recognize most of the code:
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md]:
```suggestion
Another use for stream.pipe is file streams. In Node.js, fs.createReadStream and fs.createWriteStream are used to create a stream to an open file descriptor. Now let's look at how one might use stream.pipe to write to a file. You'll probably recognize most of the code:
```

File: locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md [hunk 10]
```diff
@@ -91,4 +91,4 @@ http.createServer(function (req, res) {
 
 One could also use stream.pipe to send incoming requests to a file for logging, or to a child process, or any one of a number of other things.
 
-Hopefully this has shown you the basics of using stream.pipe to easily pass your data streams around.  It's truly a powerful little trick in node.js, and its uses are yours to explore.  Happy coding, and try not to cross your streams!
+Hopefully this has shown you the basics of using stream.pipe to easily pass your data streams around.  It's truly a powerful little trick in Node.js, and its uses are yours to explore.  Happy coding, and try not to cross your streams!
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/streams/how-to-use-stream-pipe.md]:
```suggestion
Hopefully this has shown you the basics of using stream.pipe to easily pass your data streams around. It's truly a powerful little trick in Node.js, and its uses are yours to explore. Happy coding, and try not to cross your streams!
```

File: locale/en/knowledge/advanced/streams/what-are-streams.md [hunk 11]
```diff
@@ -8,7 +8,7 @@ difficulty: 3
 layout: knowledge-post.hbs
 ---
 
-Streams are another basic construct in node.js that encourages asynchronous coding. Streams allow you to process the data as it is generated or retrieved. Streams can be readable, writeable, or both.
+Streams are another basic construct in Node.js that encourages asynchronous coding. Streams allow you to process the data as it is generated or retrieved. Streams can be readable, writeable, or both.
 
 In other words, Streams use events to deal with data as it happens, rather than only with a callback at the end.  Readable streams emit the event `data` for each chunk of data that comes in, and an `end` event, which is emitted when there is no more data. Writeable streams can be written to with the `write()` function, and closed with the `end()` function.  All types of streams emit `error` events when errors arise.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/advanced/streams/what-are-streams.md]:
```suggestion
In other words, Streams use events to deal with data as it happens, rather than only with a callback at the end. Readable streams emit the event `data` for each chunk of data that comes in, and an `end` event, which is emitted when there is no more data. Writeable streams can be written to with the `write()` function, and closed with the `end()` function. All types of streams emit `error` events when errors arise.
```

File: locale/en/knowledge/errors/what-are-the-error-conventions.md [hunk 12]
```diff
@@ -8,7 +8,7 @@ difficulty: 1
 layout: knowledge-post.hbs
 ---
 
-In node.js, it is considered standard practice to handle errors in asynchronous functions by returning them as the first argument to the current function's callback.  If there is an error, the first parameter is passed an `Error` object with all the details. Otherwise, the first parameter is null.
+In Node.js, it is considered standard practice to handle errors in asynchronous functions by returning them as the first argument to the current function's callback.  If there is an error, the first parameter is passed an `Error` object with all the details. Otherwise, the first parameter is null.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/errors/what-are-the-error-conventions.md]:
```suggestion
In Node.js, it is considered standard practice to handle errors in asynchronous functions by returning them as the first argument to the current function's callback. If there is an error, the first parameter is passed an `Error` object with all the details. Otherwise, the first parameter is null.
```

File: locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md [hunk 13]
```diff
@@ -9,7 +9,7 @@ difficulty: 1
 layout: knowledge-post.hbs
 ---
 
-Nodejs promotes an asynchronous coding style from the ground up, in contrast to many of the most popular web frameworks. There are a number of important things to be aware of when learning to write asynchronous code - otherwise, you will often find your code executing in extremely unexpected ways.  Take this (general) rule to heart:
+Node.js promotes an asynchronous coding style from the ground up, in contrast to many of the most popular web frameworks. There are a number of important things to be aware of when learning to write asynchronous code - otherwise, you will often find your code executing in extremely unexpected ways.  Take this (general) rule to heart:
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md]:
```suggestion
Node.js promotes an asynchronous coding style from the ground up, in contrast to many of the most popular web frameworks. There are a number of important things to be aware of when learning to write asynchronous code - otherwise, you will often find your code executing in extremely unexpected ways. Take this (general) rule to heart:
```

File: locale/en/knowledge/getting-started/control-flow/what-are-callbacks.md [hunk 14]
```diff
@@ -22,7 +22,7 @@ function processData () {
 
 This works just fine and is very typical in other development environments. However, if fetchData takes a long time to load the data (maybe it is streaming it off the drive or the internet), then this causes the whole program to 'block' - otherwise known as sitting still and waiting - until it loads the data. Node.js, being an asynchronous platform, doesn't wait around for things like file I/O to finish - Node.js uses callbacks.  A callback is a function called at the completion of a given task; this prevents any blocking, and allows other code to be run in the meantime.
```
Comment (Aissaoui-Ahmed) [inline locale/en/knowledge/getting-started/control-flow/what-are-callbacks.md]:
```suggestion
This works just fine and is very typical in other development environments. However, if fetchData takes a long time to load the data (maybe it is streaming it off the drive or the internet), then this causes the whole program to 'block' - otherwise known as sitting still and waiting - until it loads the data. Node.js, being an asynchronous platform, doesn't wait around for things like file I/O to finish - Node.js uses callbacks. A callback is a function called at the completion of a given task; this prevents any blocking, and allows other code to be run in the meantime.
```

File: external/survey-2018/index.html [hunk 15]
```diff
@@ -218,7 +218,7 @@ <h1>2018 Node.js User Survey Report</h1>
           <div class="section-content row">
             <ul>
               <li type="1">
-                As with last year, the typical node.js user is male, age 31 and college educated.
+                As with last year, the typical Node.js user is male, age 31 and college educated.
               </li>
               <li type="1">
                 A majority are developers (as opposed to dev managers), in small (&lt;100 employees) companies, with 5+ years of professional development experience.
```
Comment (Aissaoui-Ahmed) [inline external/survey-2018/index.html]: Ref: #2606 
```suggestion
                A majority are developers (as opposed to dev managers), in small (<100 employees) companies, with 5+ years of professional development experience.
```

**Review by Trott — DISMISSED** [review]
Some changes needed per comments. (Not sure about all of them, but some of them are definitely required.) Feel free to dismiss this "request changes" review once changes have been applied.

**Review by stevemao — APPROVED** [review]

**Review by osk2 — APPROVED** [review]

**Review by aymen94 — APPROVED** [review]

**Review by Aissaoui-Ahmed — APPROVED** [review]
LGTM

**Review by XhmikosR — APPROVED** [review]

**Comment by XhmikosR** [comment]
BTW I wouldn't remove the double spaces after a period in this PR.

It seems it is in many places; I believe some people prefer writing like that. Personally I don't agree with that, but still, it should be done in a separate PR and tackle all of the instances.
